autogen.sh

```diff
@@ -1 +1,2 @@
+git submodule update --init
```

Comment:
> I suggest to run only of if argon2 directory doesn't exist.

There is a case where the *directory* doesn't exist **and** system argon2 submodule should be used. It seems like the second sentence doesn't answer the issue in the former. Anyway, checking if directory exists doesn't change anything since `git` will do that for us.

If Instead, we were to move the init call into the `configure.ac` spec, then the call would be part of the `configure` script. Therefore, the `git` call would occur even when run from a distributed tarball where the git repository wouldn't exist. As a quick example:
```sh
$ make dist
$ tar -xzf opendht-${version}.tar.gz
$ cd opendht-${version} && ./configure
```
The last line would give error. That's the reason why I decided to put the call into `autogen.sh` since it is a shell command that belongs into the repository scripts and not in the distributed scripts as I said earlier in other comments. This is a valid point since it is part of the features of autotools which we support. One could ask "what happens if the `autogen.sh` script is run outside of the repository" to which I would answer: "this is not a supported case since it's not handled by autotools and such case would only happen if the user would pull out the files from the repository which is not the way opendht is intended to be built."